ocs_ci/framework/tests/order/a/test_a.py

```diff
@@ -0,0 +1,40 @@
+from os import environ
+from ocs_ci.framework.pytest_customization.marks import (
+    post_ocs_upgrade, post_upgrade, ocs_upgrade
+)
+
+
+@post_upgrade
+def test_1():
+    """
+    This test depends on test_e.py in folder 'e', test_z.py in folder 'z'
+    to test our internal upgrade ordering marks.
+
+    Each test within specific folder 'e' and 'z' update same environment
+    variable 'RH' with one character in specific order to test the final
+    string "Red Hat".
+
+    Any order changes will cause the environment variable to differ.
+    """
+
+    assert environ['RH'] != "Red Hat"
```

Comment:
Shouldn't we test that the value *equals* "Red Hat" in the end instead?

Reply:
That is what it is doing, It will assert if its not equals to "Red Hat"

Reply:
It check if it's not equal Red Hat.

I also would expect
```suggestion
    assert environ['RH'] == "Red Hat", f"Compiled string: {environ['RH']} is not equal to 'Red Hat'"
```

or what exactly do you constructing? `Red Hat` no?

Reply:
Its part of unit test, we can see its working, assert will be true if its not equal and we have to check its not equal(breaks order)

ocs_ci/framework/tests/order/z/test_z.py ..                              [  3%]
ocs_ci/framework/tests/order/e/test_e.py ...                             [  8%]
ocs_ci/framework/tests/order/a/test_a.py ...                             [ 14%]

Reply:
Ok

Reply:
Hmm.

Reply:
Trying `tox -e py37 -- ocs_ci/framework/tests/order -v` I get this order:

```
ocs_ci/framework/tests/order/z/test_z.py::test_8 PASSED
ocs_ci/framework/tests/order/z/test_z.py::test_7 PASSED
ocs_ci/framework/tests/order/e/test_e.py::test_6 PASSED
ocs_ci/framework/tests/order/e/test_e.py::test_5 PASSED
ocs_ci/framework/tests/order/e/test_e.py::test_4 PASSED
ocs_ci/framework/tests/order/a/test_a.py::test_3 PASSED
ocs_ci/framework/tests/order/a/test_a.py::test_2 PASSED
ocs_ci/framework/tests/order/a/test_a.py::test_1 PASSED
```

Which seem ok, but when I edit the check and use equals instead, I'm getting:

```
>       assert environ["RH"] == "Red Hat"
E       AssertionError: assert 'Red HaT' == 'Red Hat'
E         - Red HaT
E         ?       ^
E         + Red Hat
E         ?       ^
```

So it looks like `test_2` should append letter `t` instead of `T`.

Reply:
@petr-balogh you don't really need to include assert messages, as pytest assert reporting will make the problem obvious, see example above.

Reply:
the 't' was the culprit :+1: